### PR TITLE
[CIR][Basic][NFC] Add the CIR language to the Language enum

### PR DESCRIPTION
**Pull request by lanza**
Add the CIR language to the Language enum and the standard usages of it.

commit-id:fd12b2c2

### CODE REVIEW

File: clang/include/clang/Basic/LangStandard.h [hunk 1]
```diff
@@ -41,6 +41,7 @@ enum class Language : uint8_t {
   RenderScript,
   HIP,
   HLSL,
+  CIR,
```
Comment (bcardosolopes) [inline clang/include/clang/Basic/LangStandard.h:44]: Should this come after `LLVM_IR` with its own / similar comment?

**Review by bcardosolopes — APPROVED** [review]

**Comment by fhahn** [comment]
Could you remove the commit-id line from the commit message, as it doesn’t seem relevant?

**Comment by lanza** [comment]
> Could you remove the commit-id line from the commit message, as it doesn’t seem relevant?

That's [spr](https://github.com/getcord/spr)'s bookkeeping. It doesn't make it into the final commit.

**Review by AaronBallman — APPROVED** [review]
LGTM!

**Pull request merged**